packages/playwright-core/src/server/registry/index.ts

```diff
@@ -445,14 +445,7 @@ type BrowsersJSONDescriptor = {
 };
 
 function readDescriptors(browsersJSON: BrowsersJSON): BrowsersJSONDescriptor[] {
-  const headlessShells: BrowsersJSON['browsers'] = [];
-  for (const browserName of ['chromium', 'chromium-tip-of-tree']) {
```

Comment:
I believe we actually need to preserve this logic for `browsers.json` files from releases where `-headless-shell` was computed.

Reply:
On the other hand it's already a problem for 1.50 `-headless-shell` browsers today, when running `npx playwright unsinstall` from playwright <=1.49. So reverting should be fine.